### ISSUE TITLE
 Failed to construct 'AudioWorkletNode': AudioWorkletNode cannot be created: AudioWorklet does not have a valid AudioWorkletGlobalScope. Load a script via audioWorklet.addModule() first.

### ISSUE THREAD

**Issue by hieyou1** [processing/p5.js-sound#469]
This happens when I load the script.
![image](https://user-images.githubusercontent.com/31020711/80842948-8dfb5400-8bc8-11ea-91c6-1d8a57e72c92.png)

**Comment by endurance21** [comment]
@hieyou1  
please provide more info , like browser version ? , p5.js-sound version? , using web-editor ? or using local-server ?,

**Comment by hieyou1** [comment]
P5 js sound version 1.0.0 as the link says, browser version 81 on Chrome, and I am using a local server

**Comment by endurance21** [comment]
@therewasaguy 
i did some digging and here are  my observations !

According to  https://developers.google.com/web/updates/2017/12/audio-worklet
it says , we must construct the  AudioWorkletNode using `Audio Worklet object's addModule() call`

proof :

![Selection_094](https://user-images.githubusercontent.com/43696525/80849342-6346e500-8c34-11ea-9dc3-e46e334e79ea.png)


but  , as i can see in our codebase  
https://github.com/processing/p5.js-sound/blob/6a65e795e0935cb6612bc80493e6e2fb219f92d9/src/helpers.js#L331

we are creating our AudioWorkletNode  without using  `Audio Worklet object's addModule() call`  
i think that's were the trouble begins !!!!!!


@therewasaguy 
what are your suggestions ? i think we should follow , https://developers.google.com/web/updates/2017/12/audio-worklet
to implement the same !!!

**Comment by endurance21** [comment]
@hieyou1 

I have tested on my computer with same  configurations but i got no such error!  .
I think you should tell us some more specifics about  how you are using the p5.js-sound ,  so that we can reproduce the bug ! 
however ,
i can smell it as  a genuine bug as i find some differences in the implementations !! 
but still reproducing the bug will give more clear idea !!

**Comment by hieyou1** [comment]
Hmm @endurance21 heres some data. I imported them from the Cloudflare CDN (as it says in the picture, the minified versions). The error occurs when I call new p5.AudioIn(); not when the script loads. I called new p5.AudioIn() and Chrome gave me the error.

**Comment by endurance21** [comment]
@hieyou1 
yes ! now i can reproduce the bug !  :grin:

**Comment by endurance21** [comment]
@hieyou1 
you must be calling the p5.AudioIn out of setup() function ?  :confused: 
I just have jused  used it inside the setup() and wollla! it works all perfect and fine !!!
here is the small sketch i wrote locally that takes mic input and logs the amplitude to the console and it works perfectly fine ! :smiley: 
![image](https://user-images.githubusercontent.com/43696525/80850430-60022800-8c39-11ea-854f-8df2bf0a2fe7.png)

**Comment by endurance21** [comment]
@therewasaguy 
i am  curious to know the reason why it works inside of setup() and throws error if initialized and used   out of setup() ! :grinning: 

what i am guessing  it is , all enviroments setup such as setting audiocontexts and establishing AudioProcessorNode is achieved first and then it fires setup()  ??

**Comment by hieyou1** [comment]
@endurance21 Yeah; I'm not sure about that either. I tried to initialize it outside of setup() and it failed; when calling it in setup() it worked! Hmm, don't have any idea.
